src/compiler/parser.ts

```diff
@@ -4517,6 +4523,12 @@ module ts {
         }
 
         function parseClassDeclaration(fullStart: number, modifiers: ModifiersArray): ClassDeclaration {
+            // In ES6 specification, All parts of a ClassDeclaration or a ClassExpression are strict mode code
+            if (languageVersion >= ScriptTarget.ES6) {

```

Comment:
Also, can we bring this to the design meeting.  Technically those classes are broken, and won't work in the future.  Do we want to continue 'pretending' they're ok, or should we be up frong about breaking them.  Can you also possibly email us the examples of where this breaks?